### PR TITLE
add mesh simplification to onshape exporter library

### PR DESCRIPTION
**Pull request by codekansas**
clean up the mesh simplification script and incorporate it into the kol library

### CODE REVIEW

File: kol/onshape/converter.py [hunk 1]
```diff
@@ -845,10 +846,95 @@ def resolve(expression: str | None) -> float | None:
...
+            combined_mesh.save(combined_stl_file_path)
+            # Get convex hull of combined mesh, and scale to good size.
+            combined_collision = get_mesh_convex_hull(combined_mesh)
+            combined_collision = scale_mesh(combined_mesh, 0.6)
+            # Save collision mesh to filepath
+            combined_collision_stl_name = f"{new_part_name}_collision.stl"
+            collision_stl_file_path = self.mesh_dir / combined_collision_stl_name
+            combined_collision.save(collision_stl_file_path)
+            # Get combined dynamic, visual, and collision meshes.
+            print(connected_parts)
```
Comment (codekansas) [inline kol/onshape/converter.py:886]: @wuallenwu plz avoid print statements

**Review by hu-po — APPROVED** [review]
send it :rocket:

**Comment by wuallenwu** [comment]
Current action items

1. Get mesh simplification working - specifically there's an off by one error when trying to run the current script on the random.stl in the repo. (high prio)
2. Support for all link and joint types in the stompy urdf (high prio, probably very involved)
3. Script to simplify all collision meshes in a urdf (low prio, should be pretty easy)
4. Finish adding combining two inertial rpy, test merge_fixed_joints on a bigger urdf
5. Add all the scripts to the autoupload pipeline.

File: kol/requirements.txt
Comment (hu-po) [inline kol/requirements.txt:1]: maybe worth pinning these requirement versions?

File: kol/merge_fixed_joints.py
Comment (hu-po) [inline kol/merge_fixed_joints.py:1]: I like the typing, this code is complex so its definitely a good idea

File: kol/scripts/cli.py
Comment (hu-po) [inline kol/scripts/cli.py:1]: we should add sections in the README that tell the user example commands they can run. it should be easy for someone without deep knowledge to just scroll through the readme and copy paste the one command they need.

**Comment by hu-po** [comment]
@wuallenwu overall I feel like this is a good direction to take this repo. Adding types and tests and just generally cleaning it up while you add the specific requested features (afaik simplification and merging?). But tbh its just too much for me to really dig deep and understand what is going on, let alone know how to fix specific bugs.

File: kol/merge_fixed_joints.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    collision_origin = etree.SubElement(new_collision, "origin", attrib={"xyz": "0 0 0", "rpy": "0 0 0"})
+    collision_geometry = etree.SubElement(new_collision, "geometry")
+    collision_mesh = etree.SubElement(collision_geometry, "mesh", attrib={"filename": combined_collision_stl_name})
+    logger.info("Got combined meshes and dynamics.")
+
+    # Create inertial element
+    new_inertial = etree.SubElement(new_part, "inertial")
+    inertial_mass = etree.SubElement(new_inertial, "mass", attrib={"value": str(combined_dynamics.mass)})
+    inertial_inertia = etree.SubElement(new_inertial, "inertia", attrib=matrix_to_moments(combined_dynamics.inertia))
+    # Get initial rpy of parent and child
```
Comment (budzianowski) [inline kol/merge_fixed_joints.py:205]: Remove if not used

File: kol/simplify_old/utils.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
+"""Defines mesh simplification utility functions."""
```
Comment (budzianowski) [inline kol/simplify_old/utils.py:1]: remove the whole folder?

**Review by budzianowski — CHANGES_REQUESTED** [review]
See first clean up nits. This PR is already way to big so given all the upcoming action items, you should stop adding things but polish existing version and add more tests.

Every new points from the list should be another PR.

**Pull request merged**